kotlin/services/ec2/Readme.md

```diff
@@ -1,33 +1,129 @@
+## Code examples
 
-Some of these examples perform *destructive* operations on AWS resources,  such as deleting a security group. **Be very careful** when running an operation that deletes or modifies AWS resources in your account. It's best to create separate test-only resources when experimenting with these examples.
+The credential provider used in all code examples is Shared credentials. For more information, see [Using credentials](https://docs.aws.amazon.com/sdk-for-kotlin/latest/developer-guide/credential-providers.html).
 
-You will find these examples: 
+### Get started
 
-- **AllocateAddress** - Demonstrates how to allocate an elastic IP address for an Amazon EC2 instance.
-- **CreateInstance** - Demonstrates how to create an Amazon EC2 instance.
-- **CreateKeyPair** - Demonstrates how to create an Amazon EC2 key pair.
-- **CreateSecurityGroup** - Demonstrates how to create an Amazon EC2 security group.
-- **DeleteKeyPair** - Demonstrates how to delete an Amazon EC2 key pair.
-- **DeleteSecurityGroup** - Demonstrates how to delete an Amazon EC2 security group.
-- **DescribeAccount** - Demonstrates how to get information about the Amazon EC2 account.
-- **DescribeAddresses** - Demonstrates how to get information about elastic IP addresses.
-- **DescribeInstances** - Demonstrates how to get information about all the Amazon EC2 Instances associated with an AWS account.
-- **DescribeInstanceTags** - Demonstrates how to describe the specified tags for your Amazon EC2 resource.
-- **DescribeKeyPairs** - Demonstrates how to get information about all instance key pairs.
-- **DescribeRegionsAndZones** - Demonstrates how to get information about all the Amazon EC2 regions and zones.
-- **DescribeSecurityGroups** - Demonstrates how to get information about all the Amazon EC2 security groups.
-- **DescribeVPCs** - Demonstrates how to get information about all the Amazon EC2 VPCs.
-- **FindRunningInstances** - Demonstrates how to find running Amazon EC2 instances by using a filter.
-- **TerminateInstance** - Demonstrates how to terminate an Amazon EC2 instance.
+- [Hello Amazon EC2 service](https://github.com/awsdocs/aws-doc-sdk-examples/blob/main/kotlin/services/ec2/src/main/kotlin/com/kotlin/ec2/DeleteSecurityGroup.kt) (describeSecurityGroups command)
 
-To run these examples, you can setup your development environment to use Gradle to configure and build AWS SDK for Kotlin projects. For more information, 
-see [Get started with the AWS SDK for Kotlin](https://docs.aws.amazon.com/sdk-for-kotlin/latest/developer-guide/setup.html). 
+### Single actions
+
+Code excerpts that show you how to call individual service functions.
+
+- [Allocate an elastic IP address](https://github.com/awsdocs/aws-doc-sdk-examples/blob/main/kotlin/services/ec2/src/main/kotlin/com/kotlin/ec2/AllocateAddress.kt) (allocateAddress command)
```

Comment:
Are these single action examples linking to the ones in the scenario? It looks like they are linking somewhere else.